src/wasm-interpreter.h

```diff
@@ -150,10 +150,18 @@ class Indenter {
 template<typename SubType>
 class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
 protected:
-  Index maxDepth;
+  Index maxDepth = 0; // = no limit
+
+  Index maxLoopIterations = 0; // = no limit
```

Comment:
this 0 is confusing I think. while a max depth of 0 is meaningless, a max loop iterations of 0 is valid - it means to avoid any loop iterations.

how about using `std::limits<Index>::max()` (i.e., default to the maximum-sized integer)?

Reply:
While looking at this I figured that 1 iteration is somewhat analogous to traversing a block (that we give up on if it loops), which is always safe to do and in fact is the only valid preset for precompute since 2+ iterations would require a side effect, which we cannot replace.

Now, if someone wants to disable looking at blocks, they have to override `visitBlock`, and similarly one would have to override `visitLoop` instead of setting `maxLoopIterations=0`. As such I'd somewhat prefer keeping the `0` here, since `max` might be considered a valid limit. Also maps well to `maxDepth` being disabled with `0`. Wdyt?

Reply:
I see, fair enough.